src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -752,29 +757,31 @@ private List<LaunchCommand> GetInitializeCommands()
                         commands.Add(new LaunchCommand("-exec-arguments " + _launchOptions.ExeArguments));
                     }
 
-                    Func<string, Task> breakMainSuccessHandler = (string bkptResult) =>
+                    Func<Results, Task> breakMainSuccessResultsHandler = (Results bkptResult) =>
                     {
-                        int index = bkptResult.IndexOf("number=", StringComparison.Ordinal);
-                        if (index > 0)
+                        if (bkptResult.Contains("bkpt"))
                         {
-                            string trimmedInnerText = bkptResult.Substring(index).Trim('\r', '\n', '{', '}');
-                            Dictionary<string, string> dict = trimmedInnerText
-                                .Split(',')
-                                .Select(value => value.Split('='))
-                                .Where(x => x.Length == 2)
-                                .ToDictionary(x => x.First(), x => x.Last());
-
-                            if (dict.Keys.Contains("number"))
+                            ResultValue b = bkptResult.Find("bkpt");
+                            TupleValue bkpt = null;
+                            if (b is TupleValue)
+                            {
+                                bkpt = b as TupleValue;
+                            }
+                            else if (b is ValueListValue)
                             {
-                                this._entryPointBreakpoint = dict["number"];
+                                bkpt = (b as ValueListValue).Content[0] as TupleValue;
+                            }
+
+                            if (bkpt != null)
+                            {
+                                this._entryPointBreakpoint = bkpt.FindString("number");
```

Comment:
Does GDB do the `.1` thing with `number` in the multiple bind case? If so, do we need to do anything else to pre-process the number before we send it to breakpoint delete?